src/test/run-make/macos-deployment-target/Makefile

```diff
@@ -0,0 +1,16 @@
+# only-macos
+#
+# Check that a set deployment target actually makes it to the linker.
+# This is important since its a compatibility hazard. The linker will
+# generate load commands differently based on what minimum OS it can assume.
+
+include ../../run-make-fulldeps/tools.mk
+
+MACOSX_DEPLOYMENT_TARGET = 10.9
+OUT_FILE=$(TMPDIR)/with_deployment_target.dylib
+all:
+	$(RUSTC) with_deployment_target.rs -o $(OUT_FILE)
+# XXX: The check is for either the x86_64 minimum OR the aarch64 minimum (M1 starts at macOS 11).
+# They also use different load commands, so we let that change with each too. The aarch64 check
+# isn't as robust as the x86 one, but testing both seems unneeded.
+	vtool -show-build $(OUT_FILE) | $(CGREP) -e "version 10.9|minos 11.0"
```

Comment:
This will fail to detect if x86_64 macOS starts returning `minos 11.0`. It might be better to explicitly check each separately? Not sure.

Reply:
Fair point. This is smarter now and considers the compilation arch.